pkg/coordinator_etcd.go

```diff
@@ -139,16 +137,16 @@ func (e *etcdCoordinator) getOrCreateSession(sessionID string) (*sessionMeta, er
 	return &meta, nil
 }
 
-func (e *etcdCoordinator) ensureSession(sessionID string) *sfu.Session {
+func (e *etcdCoordinator) ensureSession(sessionID string) sfu.Session {
 	e.mu.Lock()
 	defer e.mu.Unlock()
 
 	if s, ok := e.localSessions[sessionID]; ok {
 		return s
 	}
 
-	s := sfu.NewSession(sessionID, e.bufferFactory, e.datachannels, e.w)
-	s.OnClose(func() {
+	s := sfu.NewSession(sessionID, e.datachannels, e.w)
+	s.(*sfu.SessionLocal).OnClose(func() {
```

Comment:
We should change this https://github.com/pion/ion-sfu/blob/master/pkg/sfu/session.go#L39 to return SessionLocal not Session and this would be unecessary

Reply:
Actually, i guess they're doing a dynamic type cast here in sfu.SFU too https://github.com/pion/ion-sfu/blob/master/pkg/sfu/sfu.go#L213

Reply:
That's the first thing I did, but then we have issue with https://github.com/pion/ion-sfu/blob/master/pkg/sfu/peer.go#L48